plugin/storage/badger/spanstore/cache_test.go

```diff
@@ -82,15 +94,169 @@ func TestOldReads(t *testing.T) {
 
 		nuTid := tid.Add(1 * time.Hour)
 
-		cache.Update("service1", "operation1", uint64(tid.Unix()))
+		cache.Update("service1", "operation1", model.SpanKindUnspecified, uint64(tid.Unix()))
 		cache.services["service1"] = uint64(nuTid.Unix())
-		cache.operations["service1"]["operation1"] = uint64(nuTid.Unix())
+		cache.operations["service1"][model.SpanKindUnspecified]["operation1"] = uint64(nuTid.Unix())
 
 		cache.populateCaches()
 
 		// Now make sure we didn't use the older timestamps from the DB
 		assert.Equal(t, uint64(nuTid.Unix()), cache.services["service1"])
-		assert.Equal(t, uint64(nuTid.Unix()), cache.operations["service1"]["operation1"])
+		assert.Equal(t, uint64(nuTid.Unix()), cache.operations["service1"][model.SpanKindUnspecified]["operation1"])
+	})
+}
+
+func TestCacheStore_GetOperations(t *testing.T) {
+	runWithBadger(t, func(store *badger.DB, t *testing.T) {
+		cache := NewCacheStore(store, 1*time.Hour, false)
+		expireTime := uint64(time.Now().Add(cache.ttl).Unix())
+		cache.services = make(map[string]uint64)
+		serviceName := "service1"
+		operationName := "op1"
+		cache.services[serviceName] = uint64(time.Now().Unix() + 1e10)
+		cache.operations = make(map[string]map[model.SpanKind]map[string]uint64)
+		cache.operations[serviceName] = make(map[model.SpanKind]map[string]uint64)
+		for i := 0; i <= 5; i++ {
+			cache.operations[serviceName][spanKinds[i]] = make(map[string]uint64)
+			cache.operations[serviceName][spanKinds[i]][operationName] = expireTime
+		}
```

Comment:
why are you filling the cache data manually instead of using its own update methods?